.github/workflows/test.yml

```diff
@@ -34,27 +34,6 @@ jobs:
             - 'requirements.txt'
             - '.coveragerc'
 
-  style:
-    name: Check code style
-    needs: changes
-    runs-on: ubuntu-latest
-    if: ${{ needs.changes.outputs.changes == 'true' }}
-    steps:
-      - uses: actions/checkout@v2
-      - name: Set up Python
-        uses: actions/setup-python@v1
-        with:
-          python-version: 3.7
-      - name: Install dependencies
-        run: |
-          python -m pip install -U pip
-          python -m pip install -U black flake8 isort
-      - name: Check the style
-        run: |
-          black -t py36 --check theano/ tests/ setup.py conftest.py
-          isort --check theano/ tests/ setup.py conftest.py
-          flake8
-
```

Comment:
they'll occur in a different github actions job, pre-commit will run in this one https://github.com/pymc-devs/Theano-PyMC/pull/208/checks?check_run_id=1499947599

Reply:
see the checks below: 
![image](https://user-images.githubusercontent.com/33491632/101195385-3f7ad000-3657-11eb-8b2c-ea69e673a89e.png)

Reply:
Yeah, we want the style checks to run first&mdash;and in the same workflow&mdash;so that they can short-circuit the tests that follow.  

It might be better to end the entire workflow after a single failure within the matrix-generated tests, but, at the very least, we want to preserve the existing style check short-circuiting.

Reply:
Sure, thanks for explaining :+1: 

OK, I've made it so the pre-commit checks are run before the tests as part of the same job

> end the entire workflow after a single failure within the matrix-generated tests

I would've thought that that would happen already, given that the tests are being run with `-x` - is that not what you're seeing?

Reply:
No, a single matrix-generated test run will fast-fail, but once such failure will not stop the other matrix-generated test runs.

Reply:
@brandonwillard I've just noticed that you have
```
    strategy:
      fail-fast: false
```
- wouldn't you want to set that to `fail-fast: true` if you want a single failing test run to stop other matrix-generated test runs?

Reply:
Yeah, that sounds like the setting we might want to change.